datafusion/core/tests/memory_limit/mod.rs

```diff
@@ -182,6 +182,24 @@ async fn merge_join() {
         .await
 }
 
+#[tokio::test]
+async fn sort_merge_join_spill() {
+    // Planner chooses MergeJoin only if number of partitions > 1
+    let config = SessionConfig::new()
+        .with_target_partitions(2)
+        .set_bool("datafusion.optimizer.prefer_hash_join", false);
+
+    TestCase::new()
+        .with_query(
+            "select t1.* from t t1 JOIN t t2 ON t1.pod = t2.pod AND t1.time = t2.time",
+        )
+        .with_memory_limit(1_000)
+        .with_config(config)
+        .with_disk_manager_config(DiskManagerConfig::NewOs)
+        .run()
+        .await
```

Comment:
I wonder how do we know if it triggers spilling or not?

Reply:
Can we check metrics?

Reply:
Yeah, that is great idea, I was overthinking how to check that file spilled to disk but metrics is much easier, I'm adding it

Reply:
Done

Reply:
@viirya I added metrics tests in `sort_merge_join.rs` like https://github.com/apache/datafusion/pull/11218/files#diff-825342e035aec56595dce761afb00dd54e3ae663a2e24ebf3a597123e636f9e2R3140

For this exact test which runs on SQL level I'm thinking if I can access metrics some how

Reply:
It doesnt seem possible to access any metrics in this case. We can rely that if test with disabled spill is failing on mem issues, then the same test with enabled spilling is passing. Hope that is enough